src/M6Web/Bundle/RedisBundle/CacheAdapters/RedisCacheItemPoolAdapter.php

```diff
@@ -0,0 +1,185 @@
+<?php
+namespace M6Web\Bundle\RedisBundle\CacheAdapters;
+
+use M6Web\Bundle\RedisBundle\Redis\Redis as BaseRedis;
+
+use Symfony\Component\Cache\CacheItem;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Cache\CacheItemInterface;
+
+/**
+ * Class RedisCacheItemPoolAdapter
+ */
+class RedisCacheItemPoolAdapter extends BaseRedis implements CacheItemPoolInterface
+{
+    /**
+     * @var \Closure
+     */
+    private $createCacheItem;
+
+    /**
+     * @var \Closure
+     */
+    private $getItemLifeTime;
+
+    /**
+     * @var bool
+     */
+    protected $transactionInProgress = false;
+
+    public function __construct($redis, $defaultLifetime = 0)
+    {
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit, $cacheExpire) use ($defaultLifetime) {
+                $cacheItem = new CacheItem();
```

Comment:
`$key` is a protected property, so it seems to be designed to be subclassed:
```php
class MyCacheItem extends CacheItem
{
   public function __construct($key, $value, $isHit, $cacheExpire, $defaultLifetime) {
    // Content of your anonymous function
  }

  public function getLifetime() {
    return $this->expiry;
  }
}
```

Then, instead of doing 
```php
($this->createCacheItem)($key, $cacheValue, $isHit, $cacheExpire);
($this->getItemLifeTime)($item);
```
You can use
```php
$item = new MyCacheItem($key, $value, $isHit, $cacheExpire, $defaultLifetime);
$item->getLifetime();
```

Did I miss something? Seems more clear for me in this way